docs/modules/authentication.md

```diff
@@ -0,0 +1,145 @@
+        strategySecretKey: injectRuntimeEnv('STRATEGY_SECRET_KEY_ENV_VARIABLE_NAME')
+      },
+      logoutRoute: { // An HTTP route for user logout.
+          path: 'logout',
+          method: 'POST'
+      }
+      routes: [ // A list of HTTP API handlers required for for the current strategy.
+        {
+          path: 'register', // The HTTP path segment after `http://app-domain.tld/rootPath/api/`.
+          method: 'POST', // The HTTP method.
+          callback: 'auth/route_register_callback.js' // The path to the API handler's definition.
+        },
+        {
+          path: 'login',
+          method: 'POST',
+          callback: 'auth/route_login_callback.js'
+        }
+      ]
+    }
+  ])
+
+  switch (launchMode) {
+    case 'dev': {
+      await watch( // Merge the configuration objects.
+        merge([defaultResolveConfig, appConfig, devConfig, authModule])
+      )
+      break
+    }
+
+    // Handle the `prod`, `cloud`, and `test:functional` run modes based on your requirements.
+  }
+})().catch(error => {
+  // eslint-disable-next-line no-console
+  console.log(error)
+})
+```
+
+The code sample below demonstrates an example implementation of a strategy constructor passed to `createAuthModule`:
+
+```js title="auth/create_strategy.js"
+import { Strategy as StrategyFactory } from 'passport-local' // Import the passport strategy.
+
+const createStrategy = (options) => ({
+  // Export a function that accepts runtime options specified in the application config.
+  factory: StrategyFactory, // The passport strategy factory.
+  options: {
+    // Custom compile-time options ...
+    failureRedirect: (error) =>
+      `/error?text=${encodeURIComponent(error.message)}`,
+    errorRedirect: (error) =>
+      `/error?text=${encodeURIComponent(error.message)}`,
+    usernameField: 'username',
+    passwordField: 'username',
+    successRedirect: null,
+    // ... runtime options (for example, secret keys)
+    ...options,
+  },
+})
+
+export default createStrategy
+```
+
+## Add the Required API Handlers
+
+In your application, you need to implement [API handlers](../api-handlers.md) for user registration, login, and other authentication-related actions. For example, the code sample below demonstrates a user registration handler.
+
+```js title="auth/route_register_callback.js"
+import jwt from 'jsonwebtoken'
+import jwtSecret from './jwt_secret' // Store your JWT secret in a secure location, for example an environment variable
+import bcrypt from 'bcrypt'
+
+// A route handler accepts `req` as the first argument. The rest of the arguments depend on the used strategy.
+// The `Local` strategy adds two arguments - `username` and `password`.
+const routeRegisterCallback = async ({ resolve }, username, password) => {
+  const { data: existingUser } = await resolve.executeQuery({ // Query a read model to check if the user already exists.
+    modelName: 'read-model-name',
```

Comment:
maybe it is better to use  'users' and 'findUser' read model and resolver names, or at least comment that it should refer to a readModel where users are stored

Reply:
Changed to 'users' and 'findUser'.